app/src/main/java/com/gmail/nowak/wjw/popularmovies/presenter/list/MovieListActivity.java

```diff
@@ -0,0 +1,310 @@
+        binding.upperErrorMessageTv.setVisibility(View.GONE);
+        binding.categoryTitleTv.setVisibility(View.VISIBLE);
+    }
+
+    /**
+     * This method should handle UI behaviour when I got response from API, but for some reason couldn't read the results.
+     * @param list
+     */
+    //todo Q?
+    private void updateUIWhenNoResults(List<MovieListItemViewData> list) {
+        if (list == null) {
+            binding.centralErrorMessageTv.setText(getString(R.string.couldnt_read_results));
+
+        } else if (list.size() == 0) {
+            binding.centralErrorMessageTv.setText(getString(R.string.no_items_found));
+        }
+        binding.centralErrorMessageTv.setVisibility(View.VISIBLE);
+        binding.mainProgressBarr.setVisibility(View.GONE);
+//        binding.sortByTitleTv.setVisibility(View.GONE);
+        movieAdapter.clearMoviesData();
+        movieAdapter.notifyDataSetChanged();
+        updateCategoryTitle();
+    }
+
+    /**
+     * Handle UI when API request onFailure or no internet connection
+     */
+    private void updateUIOnFailure() {
+        binding.centralErrorMessageTv.setText(getString(R.string.response_failure));
+        binding.centralErrorMessageTv.setVisibility(View.VISIBLE);
+        binding.mainProgressBarr.setVisibility(View.GONE);
+//        binding.sortByTitleTv.setVisibility(View.GONE);
+        movieAdapter.clearMoviesData();
+        movieAdapter.notifyDataSetChanged();
+        updateCategoryTitle();
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(final Menu menu) {
+        new MenuInflater(this).inflate(R.menu.menu_main, menu);
+//        Timber.d("onCreateOptionsMenu()");
+        setMenuItemsVisibility(menu);
+        setUpOnMenuItemClickListeners(menu);
+        return true;
+    }
+
+    /**
+     * Set visibility of menuItems
+     * @param menu
+     */
+    private void setMenuItemsVisibility(Menu menu) {
+//        Timber.d("displayedIndex = %d", displayedList);
+        for (int i = 0; i < menu.size(); i++) {
+            boolean visible = displayedTab == i ? false : true;
+            MenuItem mitem = menu.getItem(i);
+            mitem.setVisible(visible);
+//            Timber.d("menu item ndex: %d title %s, visible: %d", i, mitem.getTitle(), visible == true ? 1 : 0);
+        }
+    }
+
+    /**
+     * Set up onMenuItemClickListeners to menu's menuItems
+     * @param menu
+     */
+    public void setUpOnMenuItemClickListeners(final Menu menu) {
+        for (int i = 0; i < menu.size(); i++) {
+            MenuItem current = menu.getItem(i);
+            current.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
+                @Override
+                public boolean onMenuItemClick(MenuItem item) {
+                    Resources res = getResources();
+                    CharSequence title = item.getTitle();
+                    if (res.getString(R.string.popular_tag_title).equals(title)) {
```

Comment:
Do not  compare titles but id: 
`if(item.getItemId() == R.id.showPopular)`
Better: change also id of every menu action to format: action_show_popular (or actionShowPopular)